controllers/components/workbenches/workbenches_controller_actions.go

```diff
@@ -0,0 +1,170 @@
+	componentsv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/components/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster"
+	odhtypes "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/types"
+	odhdeploy "github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/metadata/labels"
+)
+
+func initialize(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	rr.Manifests = append(rr.Manifests, odhtypes.ManifestInfo{
+		Path:       NotebookControllerPath,
+		ContextDir: "",
+		SourcePath: "",
+	})
+	if err := odhdeploy.ApplyParams(
+		NotebookControllerPath,
+		nil,
+		map[string]string{
+			"namespace": rr.DSCI.Spec.ApplicationsNamespace,
+		},
+	); err != nil {
+		return fmt.Errorf("failed to update params.env from %s : %w", rr.Manifests[0], err)
+	}
+
+	rr.Manifests = append(rr.Manifests, odhtypes.ManifestInfo{
+		Path:       KfNotebookControllerPath,
+		ContextDir: "",
+		SourcePath: "",
+	})
+	if err := odhdeploy.ApplyParams(
+		KfNotebookControllerPath,
+		nil,
+		map[string]string{
+			"namespace": rr.DSCI.Spec.ApplicationsNamespace,
+		},
+	); err != nil {
+		return fmt.Errorf("failed to update params.env from %s : %w", rr.Manifests[1], err)
+	}
+
+	rr.Manifests = append(rr.Manifests, odhtypes.ManifestInfo{
+		Path:       NotebookImagesPath,
+		ContextDir: "",
+		SourcePath: "",
+	})
+	if err := odhdeploy.ApplyParams(
+		NotebookImagesPath,
+		nil,
+		map[string]string{
+			"namespace": rr.DSCI.Spec.ApplicationsNamespace,
+		},
+	); err != nil {
+		return fmt.Errorf("failed to update params.env from %s : %w", rr.Manifests[2], err)
+	}
+
+	return nil
+}
+
+func devFlags(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	workbenches, ok := rr.Instance.(*componentsv1.Workbenches)
+	if !ok {
+		return fmt.Errorf("resource instance %v is not a componentsv1.Workbenches)", rr.Instance)
+	}
+
+	if workbenches.Spec.DevFlags == nil {
+		return nil
+	}
+
+	// Download manifests if defined by devflags
+	// Go through each manifest and set the overlays if defined
+	// first on odh-notebook-controller and kf-notebook-controller last to notebook-images
+	if len(workbenches.Spec.DevFlags.Manifests) != 0 {
+		for i, subcomponent := range workbenches.Spec.DevFlags.Manifests {
+			if strings.Contains(subcomponent.ContextDir, "components/odh-notebook-controller") {
+				// Download subcomponent
+				if err := odhdeploy.DownloadManifests(ctx, "odh-notebook-controller/odh-notebook-controller", subcomponent); err != nil {
+					return err
+				}
+				// If overlay is defined, update paths
+				defaultKustomizePathNbc := "base"
+				if subcomponent.SourcePath != "" {
+					defaultKustomizePathNbc = subcomponent.SourcePath
+				}
+				NotebookControllerPath = filepath.Join(odhdeploy.DefaultManifestPath, "odh-notebook-controller/odh-notebook-controller", defaultKustomizePathNbc)
+
+				rr.Manifests[i].Path = odhdeploy.DefaultManifestPath
```

Comment:
The index here may cause an our of bound issues i.e. if there are 4 entries in the `devFlags` array.
The order may also cause issues as i.e. if you have only 1 entry in the `devFLags`, the  you may override the wrong entry in the manifests

Reply:
I had an incorrect assumption previously, it should be fixed now. Leaving this conversation open for now to check